### PR TITLE
add move folder delta test

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Description

Adds a delta test case for exchange incremental
backups by moving one folder into another one.
Also sets up retrieving container IDs in the test
in preparation for other test control.

## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :robot: Test

## Issue(s)

* #1966

## Test Plan

- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/exchange/service_restore.go [hunk 1]
```diff
@@ -425,10 +425,12 @@ func restoreCollection(
...
-// generateRestoreContainerFunc utility function that holds logic for creating
-// Root Directory or necessary functions based on path.CategoryType
-// Assumption: collisionPolicy == COPY
-func GetContainerIDFromCache(
+// MakeContainerDestinaion builds the destination into the container
+// at the provided path.  As a precondition, the destination cannot
+// already exist.  If it does then an error is returned.  The provided
+// containerResolver is updated with the new destination.
+// @ returns the container ID of the new destination container.
+func MakeContainerDestinaion(
```
Comment (dadams39) [inline src/internal/connector/exchange/service_restore.go:433]: This name should be updated.  I would suggest using `Create` over `Make` in the naming convention. Be sure to update the name through out the PR.

File: src/internal/events/mock/mock.go [hunk 2]
```diff
@@ -6,20 +6,20 @@ import (
 	"github.com/pkg/errors"
 )
 
-type mockBus struct {
+type MockBus struct {
```
Comment (dadams39) [inline src/internal/events/mock/mock.go:9]: change to `Bus`

File: src/internal/operations/backup_test.go [hunk 3]
```diff
@@ -31,7 +29,272 @@ import (
...
+func (mr *mockRestorer) RestoreMultipleItems(
+	ctx context.Context,
+	snapshotID string,
+	paths []path.Path,
+	bc kopia.ByteCounter,
+) ([]data.Collection, error) {
+	mr.gotPaths = append(mr.gotPaths, paths...)
+
+	return nil, nil
+}
```
Comment (dadams39) [inline src/internal/operations/backup_test.go:50]: There are several different structures being used for abstractions. A `helper_test.go` file or `mock_test.go` file would help break up this file.

File: src/internal/operations/backup_test.go [hunk 4]
```diff
@@ -31,7 +29,272 @@ import (
...
+// TODO(ashmrtn): Really need to factor a function like this out into some
+// common file that is only compiled for tests.
+func makePath(t *testing.T, elements []string, isItem bool) path.Path {
+	t.Helper()
+
+	p, err := path.FromDataLayerPath(stdpath.Join(elements...), isItem)
+	require.NoError(t, err)
+
+	return p
+}
```
Comment (dadams39) [inline src/internal/operations/backup_test.go:229]: Why would we not place this in the `testing` package?

File: src/internal/operations/backup_test.go [hunk 5]
```diff
@@ -361,265 +556,106 @@ func (suite *BackupOpSuite) TestBackupOperation_ConsumeBackupDataCollections_Pat
...
+					assert.ElementsMatch(t, test.expected, bases)
+				},
+			}
 
-	return &snapshot.Manifest{
-		Tags: map[string]string{
-			backupIDTagKey: string(backupID),
-		},
-		IncompleteReason: incompleteReason,
+			//nolint:errcheck
```
Comment (dadams39) [inline src/internal/operations/backup_test.go:646]: Swallowing this error is going to influence troubleshooting for CI. Is this temporary?

File: src/internal/operations/backup_test.go [hunk 6]
```diff
@@ -1005,13 +1041,13 @@ func (suite *BackupOpSuite) TestBackupOperation_MergeBackupDetails_AddsItems() {
...
 				{
-					Manifest: makeManifest(backup1.ID, ""),
+					Manifest: makeManifest(suite.T(), backup1.ID, ""),
 					Reasons: []kopia.Reason{
 						pathReason1,
 					},
 				},
 				{
-					Manifest: makeManifest(backup2.ID, ""),
+					Manifest: makeManifest(suite.T(), backup2.ID, ""),
```
Comment (dadams39) [inline src/internal/operations/backup_test.go:1050]: May want to consider refactoring `makeManifest` into a few constants as it is reused several times.

File: src/internal/operations/backup_test.go [hunk 7]
```diff
@@ -31,7 +29,272 @@ import (
...
+		)
+	}
+
+	return res
+}
+
+func makeManifest(t *testing.T, backupID model.StableID, incompleteReason string) *snapshot.Manifest {
+	t.Helper()
+
+	backupIDTagKey, _ := kopia.MakeTagKV(kopia.TagBackupID)
```
Comment (dadams39) [inline src/internal/operations/backup_test.go:286]: What variable is being swallowed?

File: src/internal/operations/backup_integration_test.go [hunk 8]
```diff
@@ -0,0 +1,869 @@
...
+	}
+}
+
+// ---------------------------------------------------------------------------
+// Incremental Item Generators
+// TODO: this is ripped from factory.go, which is ripped from other tests.
+// At this point, three variation of the sameish code in three locations
+// feels like something we can clean up.  But, it's not a strong need, so
+// this gets to stay for now.
+// ---------------------------------------------------------------------------
```
Comment (dadams39) [inline src/internal/operations/backup_integration_test.go:283]: +1 for creating an issue for this

File: src/internal/operations/backup_test.go (hunk 3 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go]: We can take care of that in a future cleanup.

File: src/internal/operations/backup_test.go (hunk 4 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go]: Circular dependencies, mostly, but also smells.  /Testing, in general, shouldn't produce higher level package's structs, else those packages are unable to use the /testing package due to circular imports.

File: src/internal/operations/backup_test.go (hunk 5 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go]: This code isn't changed in the PR: you're looking at bad diffs due to conflicts.

File: src/internal/operations/backup_test.go (hunk 6 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go]: This code is not part of the current PR.

File: src/internal/operations/backup_test.go (hunk 7 above)
Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go]: The tag value generated by the func.  Also, this code is not part of the current PR.

**Review by dadams39 — APPROVED** [review]
gtg

**Pull request merged**